connector/profilingmetricsconnector/general_aggregation.go

```diff
@@ -0,0 +1,142 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package profilingmetricsconnector // import "github.com/elastic/opentelemetry-collector-components/connector/profilingmetricsconnector"
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	// Breakdown of the regex:
+	// `(?:\S+\s+)?`
+	//   - `(?:\S+\s+)`: Non-capturing group for the optional return type (e.g., "void ", "int ", "long ").
+	//     - `\S+`: Matches one or more non-whitespace characters (the return type).
+	//     - `\s+`: Matches one or more whitespace characters.
+	//   - `?`: Makes the entire return type group optional.
+	//
+	// `(`
+	//   - Start of the first capturing group, which will contain the full class path.
+	//
+	// `[a-zA-Z_][a-zA-Z0-9_.$<>+]+`
+	//   - Matches the first segment of the class name (e.g., "org", "sun", "MyClass").
+	//   - `[a-zA-Z_]`: Starts with a letter or underscore (standard Java identifier start).
+	//   - `[a-zA-Z0-9_.$<>+]+`: Followed by one or more letters, numbers, underscores,
+	//     dots (for package separators), dollar signs (for inner classes),
+	//     angle brackets (`<`, `>`) and plus signs (`+`) for synthetic class names
+	//     like those generated for lambdas (e.g., `$$Lambda+<hidden>`).
+	//
+	// `(?:\.[a-zA-Z_][a-zA-Z0-9_.$<>+]+)*`
+	//   - Non-capturing group for subsequent package/class segments (e.g., ".apache.lucene.search").
+	//   - `\.`: Matches a literal dot.
+	//   - `[a-zA-Z_][a-zA-Z0-9_.$<>+]+`: Matches another identifier-like segment.
+	//   - `*`: Allows zero or more of these segments.
+	//
+	// `)`
+	//   - End of the first capturing group (the full class path).
+	//
+	// `\.`
+	//   - Matches the literal dot that separates the class name from the method name.
+	//
+	// `([a-zA-Z_][a-zA-Z0-9_.$<>+]+)`
+	//   - Second capturing group, for the method name itself.
+	//   - `[a-zA-Z_][a-zA-Z0-9_.$<>+]+`: Matches a valid method name, which can also include
+	//     `$` for synthetic methods (e.g., `lambda$fillGapRunnable$13`).
+	//
+	// `\(`
+	//   - Matches the literal opening parenthesis that signifies the start of method arguments.
+	hotspotRegex = regexp.MustCompile(`(?:\S+\s+)?([a-zA-Z_][a-zA-Z0-9_.$<>+]+(?:\.[a-zA-Z_][a-zA-Z0-9_.$<>+]+)*)\.([a-zA-Z_][a-zA-Z0-9_.$<>+]+)\(`)
```

Comment:
Since we know the frame is Hotspot interpreted, instead of using a regex isn't it simpler to split the input string by '.' and then:

`result[:-1]` will be the `fullClassName`
`result[-1]` will be the method call

If we want to get rid of the arguments, we can use this:

`result := re.sub('\(.*\) ?.*$', '', input)`

Reply:
We should first:

1. Get rid of the return type (it's always followed by a space, so splitting by `" "` and taking the element at index 1 will do it)
2. Get rid of the arguments (e.g. via simpler regex `"\(.*\) ?.*$"`)

Reply:
With https://github.com/elastic/opentelemetry-collector-components/pull/687/commits/9735400e1dacce69cf0ab1d434c0b13831c86c29 I have replaces the regex matching with functions only using the strings package. Quick benchmarks also show, that just using strings funtions performs better in terms of computing time and also reduces allocations.